### ISSUE TITLE
Extension review

### ISSUE THREAD

**Issue by MegaphoneJon** [konadave/com.klangsoft.flexiblejobs#1]
Hi Dave,

I'm on the CiviCRM extension review team.  Thanks for submitting this!

Overall, it looks good.  Here's the feedback I have.
- I struggled a bit with setting the date.  I didn't realize dates in the past wouldn't work (I was trying to use them so I could test immediately), so I thought I couldn't figure out the date format it was expecting.  I finally saw the code that was confounding me - but not until I converted that field to a date picker.  I'll submit that as a PR.
- A few minor civilint issues - but tbh, civix generates code with some of the same errors.  I'm ignoring this.
- The biggest issue is the use of the core file overrides, in particular SelectValues.php.  This makes me worry about this breaking a point release upgrade - and of course, using this with 4.7 as-is isn't an option.
- The other reason this can't be used with 4.7 is your code makes use of the group_name column in civicrm_setting, which is removed in 4.7.

So with that all said, I'm going to approve this immediately, but if you want to incorporate my PR, I think it's a worthwhile improvement! 

Finally - I think these features would be welcome in core, and would be easier to maintain than PHP overrides.  For the future, perhaps.

**Comment by konadave** [comment]
Jon,

Thanks for the date picker changes, that's probably more user friendly than an interpretive text field. I did have to make a few adjustments so that it was properly saved and for defaulting the values on edit, but it's all good now.

As for testing, I backdated the last run of a job and then invoked cron using the URL method within my browser. I don't really have any experience writing unit tests and I'm not sure this could be tested that way.

I also fixed the non civix generated civilint issues.

I really wanted to get this into core but nobody seems the least bit interested. Coleman thinks these run frequencies are redundant with the new ones added in 4.7. Eileen wanted me to get feedback on the first/next run field before submitting a PR on that.

https://issues.civicrm.org/jira/browse/CRM-17669
https://github.com/civicrm/civicrm-core/pull/7404
https://issues.civicrm.org/jira/browse/CRM-17686

I prefer not to use overrides when I can, but since I already had this implemented to be added to core, it was just easier to package the modified files into an extension. It's possible I could have used the form hooks to handle the first/next run field, but ScheduledJob.php and SelectValues.php would still have to be overridden. I'm aware that this can break things in the future, and I fully intend to update the extension when needed. The other extension I've published, Extension File Overrides (https://github.com/konadave/com.klangsoft.overrides), will let me know when that's needed. I also plan to branch and create a 4.7 version once it's released.

Unless you see something else that should be addressed, I'll update info.xml and create a new release.

Thanks for looking at and approving my extension so quickly.
